src/SqlBinding/SqlAsyncCollector.cs

```diff
@@ -234,6 +234,18 @@ public TableInformation(IEnumerable<MemberInfo> primaryKeys, IDictionary<string,
                 };
             }
 
+            /// <summary>
+            /// Generates SQL query that can be used to check if a table exists
+            /// </summary>
+            public static string GetTableExistsQuery(string schema, string tableName)
+            {
+                return $@"
+                    if (exists (select * from INFORMATION_SCHEMA.TABLES where TABLE_NAME = '{tableName}' and TABLE_SCHEMA = {schema}))
```

Comment:
Let's chat about this a bit. I'd like to minimize the number of calls to the database (while maintaining a nice customer experience with human-readable error messages 😄).

Currently, what's the error that a user sees? Is it around not being able to find a primary key, or is it some giant error message from sqlclient?

Reply:
This is the current error:
System.Private.CoreLib: Exception while executing function: TimerTriggerProducts. Microsoft.Azure.WebJobs.Host: Error while handling parameter products after function returned:. **SqlBinding: Did not retrieve any primary keys for 'Products' in schema 'SCHEMA_NAME().' Cannot generate upsert command without them.**

Maybe instead of having a new query, we can add the check to the GetColumnsDefinitionsQuery? If the table exists then return the column definitions, else return false.

Reply:
@lucyzhang929 if we can work it out where we can distinguish between:
- table doesn't exist and
- table exists but no primary key exists on table

and still use the existing query, then that'd be ideal. lemme know if you need any help!